src/tags.psm1

```diff
@@ -0,0 +1,1374 @@
+    if (-not $PSBoundParameters.ContainsKey("Verbose")) { $VerbosePreference = $PSCmdlet.GetVariableValue("VerbosePreference") }
+
+    # use the helper function Resolve-AssetPartitionId to get the asset partition id.
+    $AssetPartitionId = Resolve-AssetPartitionId -Appliance $Appliance -AccessToken $AccessToken -Insecure:$Insecure -AssetPartition $AssetPartition -AssetPartitionId $AssetPartitionId
+    $local:RelPath = "AssetPartitions/$AssetPartitionId/Tags"
+
+    $local:Body = @{
+        AssetPartitionId = $AssetPartitionId;
+        Name = "$Name";
+    }
+    if ($PSBoundParameters.ContainsKey("Description")) { $local:Body.Description = $Description }
+    if ($PSBoundParameters.ContainsKey("AssetTaggingRule")) { $local:Body.AssetTaggingRule = $AssetTaggingRule }
+    if ($PSBoundParameters.ContainsKey("AssetAccountTaggingRule")) { $local:Body.AssetAccountTaggingRule = $AssetAccountTaggingRule }    
+    if ($PSBoundParameters.ContainsKey("Owner")) {
+        if (-not (Get-Module users)) {
+            Import-module "$PSScriptRoot\users.psm1"
+        }
+        $local:Body.ManagedBy = @()
+        $Owner | ForEach-Object {
+            $local:Body.ManagedBy += (Resolve-SafeguardUserObject -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure $_)
+        }
+    }
+    Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core POST "$local:RelPath" -Body $local:Body
+}
+
+
+<#
+.SYNOPSIS
+Update an existing tag via the REST API.
+
+.DESCRIPTION
+Update an existing tag. The tag is retrieved based on the ID of tag (not the name of the tag).
+This allows the Name parameter to be used to set a new name for tag.
+Please note that changing a Tag name may impact dynamic asset groups and dynamic account groups which use "Contains" in the rule.
+If an value is set on the tag in safeguard but is not passed in as a parameter, then the value will be cleared.
+For example if the AssetTaggingRule is set on the tag but is not specified as a parameter when calling the Update-SafeguardTag function, then it will be cleared.
+
+.PARAMETER Appliance
+IP address or hostname of a Safeguard appliance.
+
+.PARAMETER AccessToken
+A string containing the bearer token to be used with Safeguard Web API.
+
+.PARAMETER Insecure
+Ignore verification of Safeguard appliance SSL certificate.
+
+.PARAMETER AssetPartition
+An integer containing an ID or a string containing the name of the asset partition where the tag should be created.
+
+.PARAMETER AssetPartitionId
+An integer value containing the asset partition ID in which the tag resides.
+
+.PARAMETER TagId
+Mandatory integer with the ID of the tag. 
+The tag is retrieved based on the ID so that changing the name is possible.
+
+.PARAMETER Name
+A mandatory string containing the name for the tag.
+
+.PARAMETER Description
+An optional string containing the description for the new tag.
```

Comment:
"...for the new tag." Should something like be "...for the updated tag", correct? Or just "...for the tag."